Bottom: 77586aea0fdb458cc8d360472bbd503381a825dd
Top:    07b417a182c3180fdbe391fe407ffdcb4c5c273c
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-04-28 14:50:03 +0100

Fixup usage of the never initialised, obsolete, Output.{Width,Height}

For the stroke handling case (relating to zooming) it isn't clear what is
correct, but for now, substitute PCB->MaxWidth and PCB->MaxHeight.

This may still leave stroke based zooming broken, but it was CERTAINLY
broken before.


---

diff --git a/src/action.c b/src/action.c
index c03042e..ed087c1 100644
--- a/src/action.c
+++ b/src/action.c
@@ -549,14 +549,18 @@ FinishStroke (void)
 	    LocationType x = (StrokeBox.X1 + StrokeBox.X2) / 2;
 	    LocationType y = (StrokeBox.Y1 + StrokeBox.Y2) / 2;
 	    int z;
+	    /* XXX: PCB->MaxWidth and PCB->MaxHeight may be the wrong
+	     *      divisors below. The old code WAS broken, but this
+             *      replacement has not been tested for correctness.
+	     */
 	    z =
 	      1 +
-	      log (fabs (StrokeBox.X2 - StrokeBox.X1) / Output.Width) /
+	      log (fabs (StrokeBox.X2 - StrokeBox.X1) / PCB->MaxWidth) /
 	      log (2.0);
 	    z =
 	      MAX (z,
 		   1 +
-		   log (fabs (StrokeBox.Y2 - StrokeBox.Y1) / Output.Height) /
+		   log (fabs (StrokeBox.Y2 - StrokeBox.Y1) / PCB->MaxHeight) /
 		   log (2.0));
 	    SetZoom (z);
